model/host/host_test.go

```diff
@@ -4829,6 +4829,112 @@ func TestSetNewSSHKeys(t *testing.T) {
 	assert.Subset(t, dbHost.SSHKeyNames, []string{"foo", "bar"})
 }
 
+func TestHandleTerminatedHostSpawnedByTask(t *testing.T) {
+	for name, testCase := range map[string]struct {
+		t                *task.Task
+		h                *Host
+		newIntentCreated bool
+		hostDetailsSet   bool
+	}{
+		"TaskStillRunning": {
+			t: &task.Task{
+				Id:        "t0",
+				Execution: 0,
+				Status:    evergreen.TaskStarted,
+			},
+			h: &Host{
+				Id: "h0",
+				SpawnOptions: SpawnOptions{
+					SpawnedByTask:       true,
+					TaskID:              "t0",
+					TaskExecutionNumber: 0,
+				},
+				Status: evergreen.HostStarting,
+			},
+			newIntentCreated: true,
+			hostDetailsSet:   false,
+		},
+		"TaskAborted": {
+			t: &task.Task{
+				Id:        "t0",
+				Execution: 0,
+				Status:    evergreen.TaskStarted,
+				Aborted:   true,
+			},
+			h: &Host{
+				Id: "h0",
+				SpawnOptions: SpawnOptions{
+					SpawnedByTask:       true,
+					TaskID:              "t0",
+					TaskExecutionNumber: 0,
+				},
+				Status: evergreen.HostStarting,
+			},
+			newIntentCreated: false,
+			hostDetailsSet:   true,
+		},
+		"HostAlreadyRuning": {
+			t: &task.Task{
+				Id:        "t0",
+				Execution: 0,
+				Status:    evergreen.TaskStarted,
+				Aborted:   true,
+			},
+			h: &Host{
+				Id: "h0",
+				SpawnOptions: SpawnOptions{
+					SpawnedByTask:       true,
+					TaskID:              "t0",
+					TaskExecutionNumber: 0,
+				},
+				Status: evergreen.HostRunning,
+			},
+			newIntentCreated: false,
+			hostDetailsSet:   true,
+		},
+		"HostNotSpawnedByTask": {
+			t: &task.Task{
+				Id: "t0",
+			},
+			h: &Host{
+				Id: "h0",
+				SpawnOptions: SpawnOptions{
+					SpawnedByTask: false,
+				},
+			},
+			newIntentCreated: false,
+			hostDetailsSet:   false,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			require.NoError(t, db.ClearCollections(Collection, task.Collection))
```

Comment:
I would defer clear the collections at the end of these tests to avoid leaving random test documents in people's DBs.

Reply:
done